include/logger.h

```diff
@@ -58,12 +68,15 @@ template <const char *Fmt> class Record {
 
 public:
   Record(std::ostream &s, ContextInfo input_info, std::mutex &Logging_lock)
-      : stream(s), info(input_info), lock(Logging_lock) {
+      : stream(s), info(input_info), lock(Logging_lock), hash(0) {
```

Comment:
I'm still a little confused - right now hash seems to be made up of the contents of what gets printed out, whereas the state seems generated from the context. If you have time could you clarify/give an example?